### PR TITLE
add -github-actions

### PR DESCRIPTION
**Pull request by wullub**

### CODE REVIEW

File: .github/workflows/plan_iis_preprod.yml [hunk 1]
```diff
@@ -63,27 +63,50 @@ jobs:
...
+
+      - name: Post a post merge PR comment with the terraform plan output
+        if: ${{ github.event.pull_request.merged || github.event_name == 'push' }}
+        uses: guchaocharlie/comment-pr@v1
+        with:
+          github-token: ${{ env.GITHUB_TOKEN }}
+          comment: "Terraform plan output in ${{ env.ENVIRONMENT }}: \n```\n${{ env.TF_PLAN_OUT }}\n```"
+
+      - name: terraform apply
+        if: ${{ github.event.pull_request.merged || github.event_name == 'push' || github.event_name == 'schedule' }}
```
Comment (reallydontask) [inline .github/workflows/plan_iis_preprod.yml:97]: Guessing this is a copy and paste as there is no schedule defined, nor indeed any push

File: .github/workflows/plan_iis_preprod.yml [hunk 2]
```diff
@@ -63,27 +63,50 @@ jobs:
...
+        run: |
+          terraform apply -auto-approve -no-color tf.plan > tf_out.txt
+          tail -c 65476 tf_out.txt > tf_apply.txt
+          cat tf_apply.txt
+          echo 'TF_APPLY_OUT<<EOF' >> $GITHUB_ENV
+          cat tf_apply.txt >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+      
+      - name: Post a post merge PR comment with the terraform apply output
+        if: ${{ github.event.pull_request.merged || github.event_name == 'push' }}
```
Comment (reallydontask) [inline .github/workflows/plan_iis_preprod.yml:108]: there is no push on your triggers so this condition is odd.

I must confess that I'm not 100% whether **github.event.pull_request.merged** is enough

File: .github/workflows/plan_iis_preprod.yml [hunk 3]
```diff
@@ -11,33 +11,33 @@ on:
...
-        run: |
-          echo "ARM_CLIENT_ID=`echo ${{ secrets.DIGITAL_STUDIO_INFRA_PROD }} | sed 's/.*clientId:\([^ ]*\).*/\1/g'`" >> $GITHUB_ENV
-          echo "ARM_CLIENT_SECRET=`echo ${{ secrets.DIGITAL_STUDIO_INFRA_PROD }} | sed 's/.*clientSecret:\([^ ]*\).*/\1/g'`" >> $GITHUB_ENV
-          echo "ARM_SUBSCRIPTION_ID=`echo ${{ secrets.DIGITAL_STUDIO_INFRA_PROD }} | sed 's/.*subscriptionId:\([^ ]*\).*/\1/g'`" >> $GITHUB_ENV
-          echo "ARM_TENANT_ID=`echo ${{ secrets.DIGITAL_STUDIO_INFRA_PROD }} | sed 's/.*tenantId:\([^ ]*\).*/\1/g'`" >> $GITHUB_ENV
-
-      # - uses: azure/login@v1
-      #   with:
-      #     creds: ${{ secrets.DIGITAL_STUDIO_INFRA_PROD }}
+      # - name: define arm vars
```
Comment (reallydontask) [inline .github/workflows/plan_iis_preprod.yml:36]: I would remove this step

File: .github/workflows/plan_iis_preprod.yml [hunk 4]
```diff
@@ -11,33 +11,33 @@ on:
...
   WORKING_DIRECTORY: iis/preprod
+  ARM_CLIENT_ID: d91544c4-212d-4da5-b6a4-ddf65fe188a5
+  ARM_CLIENT_SECRET: ${{ secrets.DIGITAL_STUDIO_INFRA_PROD_CLIENT_SECRET }}
+  ARM_SUBSCRIPTION_ID: a5ddf257-3b21-4ba9-a28c-ab30f751b383
+  ARM_TENANT_ID: 747381f4-e81f-4a43-bf68-ced6a1e14edf
 
 jobs:
   verify-terraform-config:
     runs-on: ubuntu-latest
     environment: iis/preprod
```
Comment (reallydontask) [inline .github/workflows/plan_iis_preprod.yml:33]: the environment is empty, what did you have in mind to add here?

Comment (reallydontask) [inline .github/workflows/plan_iis_preprod.yml:33]: are you not able to have a variable here?

**Review by reallydontask — CHANGES_REQUESTED** [review]
Can you  tidy up the conditions as per comments please?

File: .github/workflows/plan_iis_preprod.yml (hunk 2 above)
Comment (reallydontask) [reply .github/workflows/plan_iis_preprod.yml:108]: I think the above will work

https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#pull_request

> If the action is closed and the merged key is false, the pull request was closed with unmerged commits. If the action is closed and the merged key is true, the pull request was merged.

Comment (wullub) [reply .github/workflows/plan_iis_preprod.yml:111]: added push condition

File: .github/workflows/plan_iis_preprod.yml (hunk 4 above)
Comment (wullub) [reply .github/workflows/plan_iis_preprod.yml:42]: i like that the state of them show here https://github.com/ministryofjustice/digital-studio-infra so you can see if any of the builds are failing

Comment (wullub) [reply .github/workflows/plan_iis_preprod.yml:42]: when I tried ${{ env.ENVIRONMENT }} it didn't work

Comment (reallydontask) [reply .github/workflows/plan_iis_preprod.yml:42]: That's cool, I didn't know about that, learnt one thing today, can go home now 🤔

Comment (reallydontask) [reply .github/workflows/plan_iis_preprod.yml:42]: fair enough

File: azure-devtest.tf [hunk 5]
```diff
@@ -17,9 +17,9 @@ variable "azure_app_service_oid" {
   default = "5b2509b1-64bd-4117-b839-9b0c2b02e02c"
 }
 
-variable "azure_git_repo_app_oid" {
+variable "azure_repo_app_principle_oid" {
```
Comment (reallydontask) [inline azure-devtest.tf:20]: it's _principal_ rather than _principle_.

File: shared/modules/azure-app-service/variables.tf [hunk 6]
```diff
@@ -141,7 +141,7 @@ variable "azure_app_service_oid" {
   default = "5b2509b1-64bd-4117-b839-9b0c2b02e02c"
 }
 
-variable "azure_git_repo_app_oid" {
+variable "azure_repo_app_principle_oid" {
```
Comment (reallydontask) [inline shared/modules/azure-app-service/variables.tf:144]: This objectId is different to the other two, is this a personal SP you were using?

**Review by reallydontask — CHANGES_REQUESTED** [review]
please fix the typo

**Review by reallydontask — APPROVED** [review]

**Review by dteelingmoj — APPROVED** [review]
LGTM

**Pull request merged**